cpp/src/toppra/geometric_path/piecewise_poly_path.cpp

```diff
@@ -90,6 +173,41 @@ void PiecewisePolyPath::checkInputArgs() {
   }
 }
 
+void PiecewisePolyPath::checkInputArgs(const Vectors &positions, const Vector &times,
+        const std::array<BoundaryCond, 2> &bc_type) {
+    if (positions.size() != times.rows()) {
+        throw std::runtime_error("The length of 'positions' doesn't match the length of 'times'.");
+    }
+    if (times.rows() < 2) {
+        throw std::runtime_error("'times' must contain at least 2 elements.");
+    }
+    for (size_t i = 1; i < positions.size(); i++) {
+        if (positions[i].rows() != positions[i - 1].rows()) {
+            throw std::runtime_error("The number of elements in each position has to be equal.");
+        }
+    }
+    Vector dtimes (times.rows() - 1);
+    for (size_t i = 1; i < times.rows(); i++) {
+        dtimes(i - 1) = times(i) - times(i - 1);
+        if (dtimes(i - 1) <= 0) {
+            throw std::runtime_error("'times' must be a strictly increasing sequence.");
+        }
+    }
+
+    // Validate boundary conditions
+    int expected_deriv_size = positions[0].size();
+    for (const BoundaryCond &bc: bc_type) {
+        if (bc.order != 1 and bc.order != 2) {
```

Comment:
Although more readable, `and` can cause trouble for some compilers such as MSVC (see https://stackoverflow.com/questions/555505/when-were-the-and-and-or-alternative-tokens-introduced-in-c/555524#555524)

Reply:
Noted. Changes made in https://github.com/hungpham2511/toppra/pull/161/commits/9e71bff9480547cff375a23ae9692c4a6c132dab. Thanks